Automattic/wp-calypso
# Purchases: Handle asynchronous plan removals/cancellations from Atomic sites

Fixes https://github.com/Automattic/wp-calypso/issues/55194

#### Changes proposed in this Pull Request

<img width="984" alt="Screen Shot 2021-09-15 at 14 00 11" src="https://user-images.githubusercontent.com/1233880/133429500-9f6e9ccb-29ed-453f-8aea-68809af97bab.png">

Implements a basic proof of concept that handles the asynchronous removals/cancellations of plans from Atomic sites.

The PoC lives behind the `atomic/automated-revert` feature flag which is disabled by default in all environments.

#### Testing instructions

- Use the Calypso live link below.
- Switch to an Atomic site.
- Go to Upgrades > Purchases.
- Select the site plan.
- Try removing/cancelling it.
- Make sure you are asked to contact support in order to remove/cancel it.
- Append `?flags=atomic/automated-revert` at the end of the URL and reload.
- Go through the removal/cancellation flow.
- Make sure you get a success notification indicating the removal/cancellation has been queued.
- Make sure you cannot remove/cancel the plan again (the button shouldn't show up).
- Manually run the `atomic_subscription_revert_after_lossless_import` async job (see testing instructions in D66763-code).
- Reload Calypso.
- Make sure the Atomic site is on a Free plan now.
- Make sure there are no regression when removing/cancelling a plan from a Simple site (the success notification should denote the plan has been immediately removed/cancelled).
